devito/types/dense.py

```diff
@@ -1078,19 +1072,19 @@ def __fd_setup__(self):
     def _fd_priority(self):
         return 1 if self.staggered.on_node else 2
 
-    @property
-    def is_parameter(self):
-        return self._is_parameter
-
     def _eval_at(self, func):
-        if not self.is_parameter or self.staggered == func.staggered:
+        if self.staggered == func.staggered:
             return self
-        mapper = {self.indices_ref[d]: func.indices_ref[d]
-                  for d in self.dimensions
-                  if self.indices_ref[d] is not func.indices_ref[d]}
-        if mapper:
-            return self.subs(mapper)
-        return self
+
+        mapper = {}
```

Comment:
aka 

```
m0 = self.indices_ref
m1 = func.indices_ref

mapper = {m[d]: m1[d] for d in self.dimensions if m0.get(d) is not m1.get(d)}
```

?

Reply:
No that would still throw a KeyError when d is in `m0` and not `m1` as `m0.get(d) is not m1.get(d)` would be True but `m1[d]` doesn't exist